app/controllers/api/v1/payoffs_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Api::V1
+  class PayoffsController < ApplicationController
+    before_action :set_loan, only: %i[create]
+    def index
+      @payoffs = Payoff.all
+      render json: @payoffs.to_json(methods: %i[amount outstanding_amount],
+                                    include: %i[deal])
+    end
+
+    def show
+      @payoff = Payoff.find(params[:id])
+      render json: @payoff.to_json(methods: %i[amount outstanding_amount],
+                                   include: %i[line_items deal loan received_payments])
+    end
+
+    def destroy
+      @payoff = Payoff.destroy(params[:id])
+      render json: @payoff
+    end
+
+    def create
+      @payoff = @loan.create_payoff(params[:payoff_date])
```

Comment:
Again, you want to use payoff_params in these cases to guard from SQL injection attack.